### PR TITLE
Add syntax check tests for module examples (rebased)

### PR DESCRIPTION
**Pull request by jwpeterson**
Rebase #7331 and try to get rid of remaining timeout errors.

Refs #7319.

### CODE REVIEW

**Review by dschwen — APPROVED** [review]
👍

File: modules/phase_field/examples/rigidbodymotion/tests [hunk 1]
```diff
@@ -12,6 +12,10 @@
     type = RunApp
     input = 'AC_CH_advection_constforce_rect.i'
     cli_args = '--check-input'
+    # This example uses NonlocalKernl which contains a
```
Comment (dschwen) [inline modules/phase_field/examples/rigidbodymotion/tests]: Kernel

Comment (jwpeterson) [inline modules/phase_field/examples/rigidbodymotion/tests]: Shiiiiiiiit

Comment (jwpeterson) [inline modules/phase_field/examples/rigidbodymotion/tests]: If this last one passes I will fix it during the merge.

**Pull request merged**